swig/java/apps/GDALTestIO.java

```diff
@@ -200,4 +202,32 @@ public static void main(String[] args) throws InterruptedException
 
         System.out.println("Success !");
     }
+    
+    private static void testInt64() {
+		
+		long[] data1;
+		long[] data2;
+		
+		int xSz = 5;
+		int ySz = 2;
+		int nBands = 1;
+        Driver driver = gdal.GetDriverByName("MEM");
+        Dataset dataset = driver.Create("fred", xSz, ySz, nBands, gdalconst.GDT_Int64);
+        data1 = new long[]{1,2,3,4,5,6,7,8,9,10};
```

Comment:
just to be sure there's no truncation in the process, I'll put at least one value that doesn't fit on a smaller type, like:
```suggestion
        data1 = new long[]{1,234567890123456,3,4,5,6,7,8,9,10};
```